Remove @theforeman/vendor and add all packages from it

ShimShtein opened the pull request:
This is a fix for https://github.com/theforeman/foreman-packaging/pull/12141.

ekohl commented on packages/foreman/nodejs-datatables.net-dt/nodejs-datatables.net-dt.spec:
We have all dependencies packaged for this. Please don't ship this as bundled package.

ekohl commented on packages/foreman/nodejs-datatables.net/nodejs-datatables.net.spec:
We have all dependencies already packaged so please don't ship this as a bundled package.

ekohl commented on packages/foreman/nodejs-novnc-novnc/nodejs-novnc-novnc.spec:
This package doesn't have any dependencies so please don't ship it as bundled.

ekohl reviewed (changes requested):
I just looked at a few random packages and I think you're overusing the bundle strategy.

I also see various commits that bump a package while a commit earlier added it. It would be great to see those squashed into a single commit.

ShimShtein commented:
@ekohl We're almost there. I have removed all downgrades and addressed the comments from the previous round. It got me to the repoclosure step (which is good), but the step failed: https://ci.theforeman.org/blue/organizations/jenkins/foreman-packaging-rpm-copr-pr-test/detail/foreman-packaging-rpm-copr-pr-test/5175/pipeline/733#step-828-log-34
TL;DR:
```
[2025-06-29T11:28:03.045Z]         Error: Repoclosure ended with unresolved dependencies (1) across 1 packages.
[2025-06-29T11:28:03.045Z]         package: nodejs-datatables.net-dt-1.13.5-1.el9.noarch from repo0
[2025-06-29T11:28:03.045Z]           unresolved deps (1):
[2025-06-29T11:28:03.045Z]             npm(datatables.net) >= 1.13.4
```
This is weird, because we do package `datatables.net 1.13.5` in this pr: https://github.com/theforeman/foreman-packaging/pull/12145/files#diff-5aae02da20020c721916d70c78c3074c1d927f17a43503bc8fc9406a96631726R7

What am I missing here?

ekohl reviewed (commented):
> What am I missing here?

Our packaging isn't great at checking what the resulting repository would look like. It may very well check against the base repo with the built package added to it.

I think at some point we just have to merge this, build everything and then verify repoclosure. Once we get that to pass, we can update `foreman.spec` to use the new packages.

evgeni commented on packages/foreman/nodejs-datatables.net/nodejs-datatables.net.spec:
```diff
@@ -0,0 +1,47 @@
+
+%global npm_name datatables.net
+
+Name: %{?scl_prefix}nodejs-datatables.net
```
Once this is in, we can unbundle `nodejs-datatables.net-bs` as that only has `nodejs-datatables.net` and `nodejs-jquery` as deps, and those are both packaged then.

ShimShtein replied:
Yes, that's in https://github.com/theforeman/foreman-packaging/pull/12174 already

evgeni commented on packages/foreman/nodejs-loose-envify/nodejs-loose-envify.spec:
```diff
@@ -0,0 +1,79 @@
+ExclusiveArch: %{nodejs_arches} noarch
+
+Provides: %{?scl_prefix}npm(%{npm_name}) = %{version}
+Provides: bundled(npm(js-tokens)) = 4.0.0
```
We have js-tokens 4.0.0 packaged, no need to bundle it.

ShimShtein replied:
Can we partially-bundle a package?  We still need to bundle loose-envify

evgeni replied:
This is the loose-envify package, so no, once we unbundle js-tokens, this package is "simple".

evgeni commented on package_manifest.yaml:
```diff
@@ -376,7 +376,10 @@ ruby_core_packages_tier2:
 foreman_nodejs_packages:
   hosts:
+    nodejs-ace-builds: {}
+    nodejs-apollo-client: {}
```
We have quite a few (from previous PRs too), where this is not "correct" -- should we take it as a separate task to fix em up (then we can also fix the ones from this PR) and fix the script?

evgeni commented on packages/foreman/nodejs-axios/nodejs-axios.spec:
```diff
@@ -0,0 +1,75 @@
+
+Provides: %{?scl_prefix}npm(%{npm_name}) = %{version}
+Provides: bundled(npm(axios)) = 0.21.1
+Provides: bundled(npm(follow-redirects)) = 1.15.9
```
I was curious about this and wrote https://gist.github.com/evgeni/daf75b65470f3256bd53f8f2c861203b to identify possible other cases where we can "unbundle" things after the fact
